File: spring-webflux-sleuth/micro-service-1/src/main/java/berryj/tutorial/microservice/config/WebClientConfiguration.java
package berryj.tutorial.microservice.config;

import org.springframework.beans.factory.annotation.Value;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.web.reactive.function.client.WebClient;

@Configuration
public class WebClientConfiguration {
    @Value("${application.micro-service-2.url}")
    private String microService2Url;

    @Bean
    public WebClient microServiceWebClient() {
        return WebClient.builder().baseUrl(microService2Url).build();
    }
}

File: spring-sleuth-demo/spring-sleuth-demo-1/src/main/kotlin/berryj/demo/spring/sleuth/springsleuthdemo/config/WebClientConfiguration.kt
package berryj.demo.spring.sleuth.springsleuthdemo.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.web.reactive.function.client.WebClient

@Configuration
class WebClientConfiguration {
    @Bean
    fun webClient():WebClient = WebClient.create()
}